Validate the number of commitments and signature shares

pdyraga opened the pull request:
The number of signature shares should at least be equal to the group threshold when aggregating them. We now validate their number to produce a clear error message as soon as possible. Also, the number of commitments and signature shares should be the same so that the produced signature is valid.

pdyraga committed:
Validate the number of commitments and signature shares

The number of signature shares should at least be equal to the group threshold
when aggregating them. We now validate their number to produce a clear error
message as soon as possible. Also, the number of commitments and signature
shares should be the same so that the produced signature is valid.

lukasz-zimnoch commented on frost/coordinator.go:
```diff
@@ -64,7 +68,22 @@ func (c *Coordinator) Aggregate(
 	//      Scalar z.
 
-	// TODO: validate the number of signature shares
+	if len(signatureShares) < c.threshold {
+		return nil, fmt.Errorf(
+			"not enough shares; has [%d] for threshold [%d]",
+			len(signatureShares),
+			c.threshold,
+		)
+	}
```
I would love to understand how this relates to the docstring section where we say:
```
- sig_shares, a set of signature shares z_i, Scalar values, for each
  participant, of length NUM_PARTICIPANTS, where
  MIN_PARTICIPANTS <= NUM_PARTICIPANTS <= MAX_PARTICIPANTS.
```
Do we enforce `MIN_PARTICIPANTS`/`MAX_PARTICIPANTS` or just rely on the `threshold`?

pdyraga replied:
It is the `MIN_PARTICIPANTS <= NUM_PARTICIPANTS` condition but... why not check the `<= NUM_PARTICIPANTS <= MAX_PARTICIPANTS` condition as well? Added in 375abab6e84a3e031abc5366963fa15155757386.

lukasz-zimnoch reviewed (approved):
One question, otherwise LGTM!

pdyraga committed:
Validate if the number of shares and commitments is below group size

In the previous commit, a validation whether the number is not below the
threshold was added. After adding the validation whether the number is
not higher than the group size, we do a full coverage of the
MIN_PARTICIPANTS <= NUM_PARTICIPANTS <= MAX_PARTICIPANTS condition
from the FROST paper.
